### ISSUE TITLE
scatter_add_() docs incorrectly state indices need to be unique

### ISSUE THREAD

**Issue by pimdh** [pytorch/pytorch#27080]
## 📚 Documentation

`Tensor.scatter_add_` states that the indices need to be unique, but this appears not to be the case (note that this comment is missing from `gather()` docs):
> Moreover, as for gather(), the values of index must be between 0 and self.size(dim) - 1 inclusive, and all values in a row along the specified dimension dim must be unique.

However:
```
import torch
x = torch.zeros(5)
x.scatter_add_(0, torch.tensor([0, 0, 4]), torch.tensor([1., 3., 5.]))
print(x) . # => [4, 0, 0, 0, 5]
```

Also, if the indices need to be unique, `scatter_add_` would be an incorrect backwards pass for `gather()`, whose indices do not need to be unique, according to current docs.

Docs at:
https://pytorch.org/docs/stable/tensors.html#torch.Tensor.scatter_add_

**Comment by ailzhang** [comment]
Hi @pimdh  would you like to send a PR to fix the doc? Thanks!
